### ISSUE TITLE
False Positive on a Enum.reduce call.

### ISSUE THREAD

**Issue by hickscorp** [rrrene/credo#804]
Somewhere in our code, we have something like this:

```
    ...
    |> Enum.reduce(count > 0, fn
      %{type: :individual} = onboard, true ->
        onboard
        |> Onboard.cancel_changeset()
        |> Ecto.Changeset.change(recipient_id: linked_email.user.id)
        |> repo.update() && true

      %{type: type} = onboard, seen ->
        onboard |> Ecto.Changeset.change(recipient_id: linked_email.user.id) |> repo.update()
        seen || type == :individual
    end)
```

Credo is tripping on the `Enum.reduce` line with the following:

```
[W] ↗ There should be no unused return values for Enum functions.
```

Maybe it's a false positive?

**Comment by rrrene** [comment]
Depends on what you are doing with the result.

The following does not raise an error:

```elixir
    defmodule CredoSampleModule do
      def some_function(value) do
        value
        |> Enum.reduce(count > 0, fn
          %{type: :individual} = onboard, true ->
            onboard
            |> Onboard.cancel_changeset()
            |> Ecto.Changeset.change(recipient_id: linked_email.user.id)
            |> repo.update() && true

          %{type: type} = onboard, seen ->
            onboard |> Ecto.Changeset.change(recipient_id: linked_email.user.id) |> repo.update()
            seen || type == :individual
        end)
      end
    end
```

Sometimes you want to use the reducers memo parameter, but not use the result of the `Enum.reduce/2` call (judiging from your code snippet that might be the case?).

In those cases, your only option is to disable Credo for this line using a config comment (`# credo:disable-for-next-line`) or to ignore the `reduce` function in Credo's config:

```elixir
{Credo.Check.Warning.UnusedEnumOperation, [ignore: [:reduce]]},
```

That said, this might still be a false positive. Could you share more of the surrounding function?

**Comment by rrrene** [comment]
@hickscorp *ping*
